noir/aztec_macros/src/lib.rs

```diff
@@ -390,6 +426,208 @@ fn transform_module(
     Ok(has_transformed_module)
 }
 
+/// Auxiliary function to generate constructors in the storage struct for Maps, using
+/// the Storage struct definition as a reference. Supports nesting.
+fn generate_map_field_constructor(typ: UnresolvedTypeData) -> Result<Expression, AztecMacroError> {
+    match &typ {
+        UnresolvedTypeData::Named(path, generics) => {
+            let mut new_path = path.clone().to_owned();
+            new_path.segments.push(ident("new"));
+            match path.segments.last().unwrap().0.contents.as_str() {
+                "Map" => Ok(call(
+                    variable_path(new_path),
+                    vec![
+                        variable("context"),
+                        variable("slot"),
+                        expression(ExpressionKind::Lambda(Box::new(Lambda {
```

Comment:
could be worth making a lambda helper function to collapse this a bit

Reply:
Done! Also simplified the recursive map generator to a single function.